tensorboard/plugins/graph/graph_util.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for graph plugin."""
+from tensorboard.compat.proto import function_pb2
+
+
+def combine_graph_defs(to_proto, from_proto):
+  """Combines two GraphDefs by adding nodes from from_proto into to_proto.
```

Comment:
Related to below comment, but: what does it mean to “combine” two
`GraphDef`s? If I have graphs

```
a -> b
```

and

```
a -> c -> b
```

is the result

```
 .----> b
a       ^
 `-> c -'
```

or what?

Reply:
Resolved: key sets must be disjoint.

Reply:
I was incorrect. Our hierarchy builder or graphlib (cannot tell who is doing this atm) seems to combine the NodeDef like this:
![image](https://user-images.githubusercontent.com/2547313/53263220-a0593400-368d-11e9-932a-3ce51440d477.png)